x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -472,35 +493,46 @@ public static boolean verifyHash(SecureString data, char[] hash) {
     }
 
     private static boolean verifyPbkdf2Hash(SecureString data, char[] hash) {
+        // Base64 string length : (4*(n/3)) rounded up to the next multiple of 4 because of padding,  i.e. 44 for 32 bytes
+        final int tokenLength = 44;
+        char[] hashChars = new char[tokenLength];
```

Comment:
lets set these to null. We never use these arrays. We can have null checks in the finally block.